src/types.ts

```diff
@@ -45,10 +45,10 @@ export interface NodeFileTraceOptions {
   ts?: boolean;
   log?: boolean;
   mixedModules?: boolean;
-  readFile?: (path: string) => Buffer | string | null;
-  stat?: (path: string) => Stats | null;
-  readlink?: (path: string) => string | null;
-  resolve?: (id: string, parent: string, job: Job, cjsResolve: boolean) => string | string[];
+  readFile?: (path: string) => Buffer | string | null | Promise<Buffer | string | null>;
+  stat?: (path: string) => Stats | null | Promise<Stats | null>;
+  readlink?: (path: string) => string | null | Promise<Stats | null>;
+  resolve?: (id: string, parent: string, job: Job, cjsResolve: boolean) => string | string[] | Promise<string | string[]>;
```

Comment:
If you’re worried about a breaking change, I would be fine with making these all promise return types since it’s only braking TS users. Awaiting a sync function will still work for example.

Reply:
Ah ok yeah was worried about unexpected type change, will update to just returning Promises

Reply:
Updated in https://github.com/vercel/nft/pull/236/commits/66bee8668c17d65dacf39bc05002c9943b97386f